tests/System.IO.Pipelines.Tests/SchedulerFacts.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipelines.Tests
+{
+    public class SchedulerFacts
+    {
+        [Fact]
+        public async Task ReadAsyncCallbackRunsOnSchedulerThread()
```

Comment:
Oh that is broken then. I need this code to only affect the consumer not the producer. The producer is using Task so you'll get TPL behavior.